client/src/components/Ticketing/ticketingmanager/Doorlistpage/doorlist.tsx

```diff
@@ -82,86 +86,189 @@ const DoorList = () => {
+        let deduplicatedEvents = uniqueEventIds.map((id) => jsonData.find((event) => event.eventid === id));
+
+        // Sort the events in alphabetical order by eventname
+        deduplicatedEvents = deduplicatedEvents.sort((a, b) => a.eventname.localeCompare(b.eventname));
+
+        setEventList(deduplicatedEvents);
+        setEventListFull(jsonData);
+        setEventListActive(deduplicatedEvents.filter((event) => event.active));
+      } catch (error) {
+        console.error(error.message);
+      }
+    };
+    fetchEvents();
   }, []);
 
+  const handleEventChange = (event) => {
+    const eventId = parseInt(event.target.value);
+    setSelectedEventId(eventId);
+
+    const matchingEvents = eventListFull.filter((e) => e.eventid === eventId);
+    setAvailableTimesEvents(matchingEvents);
+  };
+
+  const handleTimeChange = (event) => {
+    const eventInstanceID = parseInt(event.target.value);
+    getDoorList(eventInstanceID);
+  };
+
+  const CustomToolbar = () => {
+    const apiRef = useGridApiContext();
+
+    const handleExport = (options: any) => {
+      apiRef.current.exportDataAsCsv(options);
+    };
+
+    return (
+      <div className='container columns-3 gap-4 ml-3 mt-3 '>
+        <button
+          className='flex flex-row  text-blue-600 gap-2
+          items-center hover:text-blue-500 px-3 py-2 rounded-lg'
+          onClick={handleExport}
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
+            <path strokeLinecap="round" strokeLinejoin="round"
+              d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M9 19l3 3m0 0l3-3m-3 3V10" />
+          </svg>
+          Export
+        </button>
+      </div>
+    );
+  };
 
   const getDoorList = async (event) => {
     try {
       const token = await getAccessTokenSilently({
         audience: process.env.REACT_APP_ROOT_URL,
         scope: 'admin',
       });
-
-      const getuser = event.target.value;
-      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${getuser}`, {
+      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${event}`, {
         headers: {
-          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
         },
       });
-      const jsonData = await response.json();
-      console.log(jsonData.data);
-
-      // doorlistData.data {id: custid, name, vip, donor: donorbadge, accomodations: seatingaccom, num_tickets, checkedin, ticketno }
-      setDoorList(jsonData.data);
-      setEventName(jsonData.data[0].eventname);
-      setDate(dayMonthDate(jsonData.data[0].eventdate));
-      setTime(militaryToCivilian(jsonData.data[0].eventtime));
+      const jsonRes = await response.json();
```

Comment:
Same as above regarding adding a check to see if response.ok is true and changing variable name to be more specific.